RepositoryCompilerTests/Unit/CaDETCodeModelTests.cs

```diff
@@ -125,5 +126,34 @@ public void Determines_if_is_data_class()
             doctor.IsDataClass().ShouldBeTrue();
             service.IsDataClass().ShouldBeFalse();
         }
+
+        [Fact]
+        public void Calculates_number_of_parameters()
+        {
+            CodeModelBuilder builder = new CodeModelBuilder(LanguageEnum.CSharp);
+
+            List<CaDETClass> classes = builder.BuildCodeModel(_testDataFactory.GetGitAdapterClassText());
+
+            var gitClass = classes.First();
+            gitClass.Methods.Find(method => method.Name.Equals("CheckForNewCommits")).Metrics.NOP.ShouldBe(0);
```

Comment:
We should create a helper function for finding methods that makes the code more DRY, e.g.

FindMethod(CaDETClass c, String methodName)
{
  return c.Methods.Find(method => method.Name.Equals(methodName));
}

We can then resolve Feature Envy and reduce the parameter count by moving the method to the CaDETClass.